SplunkRumWorkspace/SplunkRum/SplunkRum/SessionBasedSampler.swift

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2021 Splunk Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import Foundation
+import OpenTelemetrySdk
+import OpenTelemetryApi
+
+class SessionBasedSampler {
+    static var probability: Double = 1.0
+    static var timer: Timer?
```

Comment:
`timer` doesn't need to be here anymore.  I don't see any subscription to `addSessionIdCallback` / logic in `Session.swift` to ensure that this re-rolls on each new session?

Reply:
@johnbley I added a callback in the init that I think will get called each time a new sessionId is generated.